src/main/java/gregtech/api/util/FluidTooltipUtil.java

```diff
@@ -1,57 +1,91 @@
 package gregtech.api.util;
 
+import com.google.common.collect.Lists;
+import com.mojang.realmsclient.gui.ChatFormatting;
 import gregtech.api.unification.material.Materials;
 import gregtech.api.unification.stack.MaterialStack;
 import net.minecraftforge.fluids.Fluid;
 import net.minecraftforge.fluids.FluidRegistry;
 import net.minecraftforge.fluids.FluidStack;
 
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 public class FluidTooltipUtil {
 
     /**
      * Registry Mapping of <Fluid, Tooltip>
      */
-    private static final Map<Fluid, String> tooltips = new HashMap<>();
+    private static final Map<Fluid, List<String>> tooltips = new HashMap<>();
 
     /**
      * Used to register a tooltip to a Fluid. A Fluid can only have one tooltip, on one line.
      *
      * Ignores a tooltip applied for Water, so that it will be handled correctly for the chemical formula.
      *
      * @param fluid   The fluid to register a tooltip for.
-     * @param tooltip The tooltip.
+     * @param tooltips The tooltip.
      * @return        False if either parameter is null or if tooltip is empty, true otherwise.
      */
+    public static boolean registerTooltip(Fluid fluid, List<String> tooltips) {
+        if (fluid != null && tooltips != null && !tooltips.isEmpty()) {
+            for (String tooltip : tooltips) {
+                registerTooltip(fluid, tooltip);
+            }
+        }
+        return false;
+    }
+
     public static boolean registerTooltip(Fluid fluid, String tooltip) {
-        if (fluid != null && tooltip != null && !tooltip.isEmpty()) {
-            tooltips.put(fluid, tooltip);
+        if (fluid != null && tooltip != null && !tooltip.trim().isEmpty()) {
+            if(tooltips.containsKey(fluid)) {
+                tooltips.get(fluid).add(tooltip);
+            } else {
+                tooltips.put(fluid, Lists.newArrayList(tooltip));
+            }
             return true;
         }
         return false;
     }
 
+
+
     /**
      * Used to get a Fluid's tooltip.
      *
      * @param fluid The Fluid to get the tooltip of.
      * @return      The tooltip.
      */
-    public static String getFluidTooltip(Fluid fluid) {
+    public static List<String> getFluidTooltips(Fluid fluid) {
         if (fluid == null)
             return null;
 
         return tooltips.get(fluid);
     }
 
+    @Deprecated
+    public static String getFluidTooltip(Fluid fluid) {
+        if (fluid == null)
+            return null;
+
+        return tooltips.get(fluid).get(1);
```

Comment:
Shouldn't this be `.get(0)`? From how we are registering it in `registerTooltip(Fluid, String)`, it does not appear to be skipping the 0 index

Reply:
I am too bit concerned why we are using sometimes 0, sometimes 1 when working with tooltips.

Reply:
I believe it is because when applying the tooltip to an ItemStack, the name of the Item is in index 0 in the list. For fluid tooltips, since I had to do it a bit more roundabout, index 0 of the registry will be put at index 1 of an ItemStack containing this fluid. Otherwise, when given to a FluidStack in JEI, it will just simply add to the end of the tooltip list rather than insert at an index.